ecosystem/typescript/sdk/src/tests/e2e/provider.test.ts

```diff
@@ -3,26 +3,46 @@ import { Network, NetworkToIndexerAPI, NetworkToNodeAPI } from "../../utils";
 
 describe("Provider", () => {
   it("uses provided network as API", async () => {
-    const provider = new Provider(Network.DEVNET);
-    expect(provider.aptosClient.nodeUrl).toBe(NetworkToNodeAPI[Network.DEVNET]);
-    expect(provider.indexerClient.endpoint).toBe(NetworkToIndexerAPI[Network.DEVNET]);
+    const providerDevnet = new Provider(Network.DEVNET);
+    expect(providerDevnet.aptosClient.nodeUrl).toBe(NetworkToNodeAPI[Network.DEVNET]);
+    expect(providerDevnet.indexerClient?.endpoint).toBe(NetworkToIndexerAPI[Network.DEVNET]);
+
+    const providerTestnet = new Provider(Network.TESTNET);
+    expect(providerTestnet.aptosClient.nodeUrl).toBe(NetworkToNodeAPI[Network.TESTNET]);
+    expect(providerTestnet.indexerClient?.endpoint).toBe(NetworkToIndexerAPI[Network.TESTNET]);
+
+    const providerMainnet = new Provider(Network.MAINNET);
+    expect(providerMainnet.aptosClient.nodeUrl).toBe(NetworkToNodeAPI[Network.MAINNET]);
+    expect(providerMainnet.indexerClient?.endpoint).toBe(NetworkToIndexerAPI[Network.MAINNET]);
   });
 
   it("uses custom endpoints as API", async () => {
     const provider = new Provider({ fullnodeUrl: "full-node-url", indexerUrl: "indexer-url" });
     expect(provider.aptosClient.nodeUrl).toBe("full-node-url/v1");
-    expect(provider.indexerClient.endpoint).toBe("indexer-url");
+    expect(provider.indexerClient?.endpoint).toBe("indexer-url");
+  });
+
+  it("does not set indexer client when indexer url is not provided with a custom network", async () => {
+    const provider = new Provider({ fullnodeUrl: "full-node-url" });
+    expect(provider.aptosClient.nodeUrl).toBe("full-node-url/v1");
+    expect(provider.indexerClient).toBe(undefined);
```

Comment:
I wonder is there some way we can make this more obvious? Seeing that it is undefined isn't super obvious, maybe some error or explanation somehow? Idk if TS has a great way to do this though, so not a blocker.

Reply:
interesting! I guess it is not really an error since we can use is without an indexer client. I'll add a note on this on the Provider class itself